ConsumerApi.Tests.Integration/Features/Devices/POST.feature

```diff
@@ -0,0 +1,11 @@
+@Integration
+Feature: POST Device
+
+User creates a Device
+
+	Scenario: Creating a Device
+		Given Identity i
+		And a Challenge c created by i
+		When i sends a POST request to the /Devices endpoint
```

Comment:
I tried to define steps in such a way so that they can be reused as often as possible. I have the `And a Challenge c created by i` step in two feature files and that step does  just that. 

If I would add `...with a valid signature` to the `WHEN` step as you suggest, then I would have to create the challenge in that step as well, and that would be duplicating the code that already does in the step definition matching the `And a Challenge c created by i` step (even if the code was extracted into a method and only the method call is duplicated).

Your call, I personally prefer it this way as it minimizes code duplication, even though semantically your option sounds better.

Reply:
But now, when you read the tests, you may wonder "Why is it required to create a challenge?". In my opinion the readability of the tests outweighs the reduction of duplication.
Plus, I'd count the kind of duplication in this case as [accidental duplication](https://medium.com/@yukas/accidental-duplication-a9760168a2aa).

Reply:
Changes made.